skytap/network.go

```diff
@@ -0,0 +1,213 @@
+type VPN struct {
+	ID            *string `json:"id"`
+	Name          *string `json:"name"`
+	Enabled       *bool   `json:"enabled"`
+	NatEnabled    *bool   `json:"nat_enabled"`
+	RemoteSubnets *string `json:"remote_subnets"`
+	RemotePeerIP  *string `json:"remote_peer_ip"`
+	CanReconnect  *bool   `json:"can_reconnect"`
+}
+
+// Tunnel is a list of connections between this network and other networks
+type Tunnel struct {
+	ID            *string  `json:"id"`
+	Status        *string  `json:"status"`
+	Error         *string  `json:"error"`
+	SourceNetwork *Network `json:"source_network"`
+	TargetNetwork *Network `json:"target_network"`
+}
+
+// CreateNetworkRequest describes the create the environment data
+type CreateNetworkRequest struct {
+	Name        *string      `json:"name"`
+	NetworkType *NetworkType `json:"network_type"`
+	Subnet      *string      `json:"subnet"`
+	Domain      *string      `json:"domain"`
+	Gateway     *string      `json:"gateway,omitempty"`
+	Tunnelable  *bool        `json:"tunnelable,omitempty"`
+}
+
+// UpdateNetworkRequest describes the update the environment data
+type UpdateNetworkRequest struct {
+	Name       *string `json:"name,omitempty"`
+	Subnet     *string `json:"subnet,omitempty"`
+	Domain     *string `json:"domain,omitempty"`
+	Gateway    *string `json:"gateway,omitempty"`
+	Tunnelable *bool   `json:"tunnelable,omitempty"`
+}
+
+// NetworkType is the type of network
+type NetworkType string
+
+// The architecture types
+const (
+	NetworkTypeAutomatic NetworkType = "automatic"
+	NetworkTypeManual    NetworkType = "manual"
+)
+
+// NetworkListResult is the listing request specific struct
+type NetworkListResult struct {
+	Value []Network
+}
+
+// List the networks
+func (s *NetworksServiceClient) List(ctx context.Context, environmentID string) (*NetworkListResult, error) {
+	path := fmt.Sprintf(networksBasePath, environmentID)
+
+	req, err := s.client.newRequest(ctx, "GET", path, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	err = s.client.setRequestListParameters(req, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var networkListResponse NetworkListResult
+	_, err = s.client.do(ctx, req, &networkListResponse.Value)
+	if err != nil {
+		return nil, err
+	}
+
+	return &networkListResponse, nil
+}
+
+// Get a network
+func (s *NetworksServiceClient) Get(ctx context.Context, environmentID string, id string) (*Network, error) {
+	path := fmt.Sprintf(networksBasePath+"/%s", environmentID, id)
+
+	req, err := s.client.newRequest(ctx, "GET", path, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var network Network
+	_, err = s.client.do(ctx, req, &network)
+	if err != nil {
+		return nil, err
+	}
+
+	return &network, nil
+}
+
+// Create a network
+func (s *NetworksServiceClient) Create(ctx context.Context, environmentID string, opts *CreateNetworkRequest) (*Network, error) {
+	path := fmt.Sprintf(networksBasePath, environmentID)
+
+	automatic := NetworkTypeAutomatic
```

Comment:
Pointer go limitation can be easy address,
Add a function to the type to do generate a literal.

Reply:
done